runtime/src/main/java/org/corfudb/runtime/view/Layout.java

```diff
@@ -398,6 +405,11 @@ public void validateSegmentSeal(LayoutSegment layoutSegment,
                 SealServersHelper.waitForChainSegmentSeal(layoutSegment, completableFutureMap);
             }
 
+            @Override
+            public int getMinReplicationFactor(Layout layout) {
+                return 2;
```

Comment:
maybe we need to have this parameterized. I am not sure what this does when there is only one node in the cluster.

Reply:
I guess then it can't tolerate any failures?

Reply:
attempting to remove a node from a one node cluster can be rejected in many places. For the base of the builder, a remove would cause (layoutStripe.getLogServers().size() < minReplicationFactor) to throw an exception. You lose replication if 1 out of 2 dies.